web/modules/custom/rdf_ff/src/SchemaFieldValidator.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Drupal\rdf_ff;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\rdf_entity\Database\Driver\sparql\Connection;
+use Drupal\Core\Entity\EntityTypeManager;
+use Drupal\rdf_entity\Entity\Query\Sparql\SparqlArg;
+use Drupal\rdf_entity\RdfEntityMappingInterface;
+use Drupal\rdf_entity\RdfFieldHandler;
+
+/**
+ * A service that validates that fields are defined in a schema.
+ */
+class SchemaFieldValidator implements SchemaFieldValidatorInterface {
+
+  /**
+   * The database connection.
+   *
+   * @var \Drupal\rdf_entity\Database\Driver\sparql\Connection
+   */
+  protected $sparqlEndpoint;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManager
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The rdf field handler service.
+   *
+   * @var \Drupal\rdf_entity\RdfFieldHandler
+   */
+  protected $fieldHanlder;
+
+  /**
+   * Constructs a new SchemaFieldValidator object.
+   *
+   * @param \Drupal\rdf_entity\Database\Driver\sparql\Connection $sparql_endpoint
+   *   The database connection.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\rdf_entity\RdfFieldHandler $field_hanlder
+   *   The rdf field handler service.
+   */
+  public function __construct(Connection $sparql_endpoint, EntityTypeManagerInterface $entity_type_manager, RdfFieldHandler $field_hanlder) {
+    $this->sparqlEndpoint = $sparql_endpoint;
+    $this->entityTypeManager = $entity_type_manager;
+    $this->fieldHanlder = $field_hanlder;
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @throws \Exception
```

Comment:
Yep, we cannot deviate from the documentation defined in the public interface, since this is the public "contract" that we are fulfilling. We shouldn't throw exceptions here if this is not defined in the interface.

If you do need to throw exceptions then an acceptable workaround is to throw a `\RuntimeException`. These are problems that are not bugs in the calling code but are happening because of problems in runtime (e.g. if we lose the connection to a remote database this is a runtime error). Runtime exceptions don't need to be documented.